rules/ruleset.go

```diff
@@ -1140,18 +1151,10 @@ type RuleSetSnapshot struct {
 	Namespace    string
 	Version      int
 	CutoverNanos int64
-	MappingRules []*MappingRuleView
-	RollupRules  []*RollupRuleView
+	MappingRules map[string]*MappingRuleView
+	RollupRules  map[string]*RollupRuleView
 }
 
-// RuleConflictError is returned when a rule modification is made that would conflict with the current state.
-type RuleConflictError struct {
-	ConflictRuleUUID string
-	msg              string
-}
-
-func (e RuleConflictError) Error() string { return e.msg }
-
 func (rs ruleSet) validateMappingRuleUpdate(mrv MappingRuleView) error {
```

Comment:
Don't we not need these functions now? Should m3ctl just call r2.Validate at the end?

Reply:
Possibly, the logic that checks for conflicting rollup targets need to be merged into the validator but yeah can remove after that's done.

Reply:
Done